workflows/phylorthology.nf

```diff
@@ -48,66 +70,416 @@ include { INPUT_CHECK } from '../subworkflows/local/input_check'
+    .branch {
+        meta, prots ->
+            proteomes  : meta.mcl_test == 'false'
+                return [ meta, prots.flatten() ]
+    }
+    .set { ch_prots_full }
+    
+    ch_prots_all
+    .branch {
+        meta, prots ->
+            proteomes  : meta.mcl_test == 'true'
+                return [ meta, prots.flatten() ]
+    }
+    .set { ch_prots_test }
+    
+    ch_prots_full
+    .branch {
+        meta, prots ->
+            proteomes  : prots
+                return [ meta, prots.flatten() ]
+    }
+    .set { ch_busco_full }
+
+    ch_prots_test
+    .branch {
+        meta, prots ->
+            proteomes  : prots
+                return [ meta, prots.flatten() ]
+    }
+    .set { ch_busco_test }
+    
+    // Create an orthofinder channel with paths to the new fasta/diamond DBs
+    ch_orthofinder_full = ch_busco_full.merge(ch_fa, ch_dmd)
+    ch_orthofinder_test = ch_busco_test.merge(ch_test_fa, ch_test_dmd)
     ch_versions = ch_versions.mix(INPUT_CHECK.out.versions)
 
+    // Now, there will be a couple of modules below that we reapply, both to 
+    // the full dataset, and to the MCL inflation parameter test set. 
+    // These repeat modules include:
+    // uniprot annotation (ch_annotations)
+    // diamond blastp (ch_blastp & ch_similarities)
+    
+    //
+    // MODULE: Annotate UniProt Proteins
+    //
+    ch_annotations = ANNOTATE_UNIPROT (
+            ch_busco_full
+        )
+        .annotations
+        .collect()
+    ch_versions = ch_versions.mix(ANNOTATE_UNIPROT.out.versions)
+    
+    //
+    // MODULE: Run BUSCO
+    // Split up into shallow and broad scale runs, since downstream modules
+    // do not use these outputs, so multiple busco runs may be conducted
+    // simultaneously
+    //
+    // Shallow taxonomic scale:
+    BUSCO_SHALLOW (
+        ch_busco_full,
+        "shallow",
+        [],
+        []
+        )
+    
+    // Broad taxonomic scale (Eukaryotes)
+    BUSCO_BROAD (
+        ch_busco_full,
+        "broad",
+        [],
+        []
+        )
+        
     //
-    // MODULE: Run FastQC
+    // MODULE: All-v-All diamond/blastp
     //
-    FASTQC (
-        INPUT_CHECK.out.reads
+    // Run for the test set (used to determine the best value of the MCL
+    // inflation parameter)
+    ch_blastp_test = DIAMOND_BLASTP_TEST (
+        ch_orthofinder_test,
+        ch_test_dmd,
+        "txt",
+        "true",
+        []
     )
-    ch_versions = ch_versions.mix(FASTQC.out.versions.first())
-
-    CUSTOM_DUMPSOFTWAREVERSIONS (
-        ch_versions.unique().collectFile(name: 'collated_versions.yml')
+    .txt
+    
+    // And for the full dataset, to be clustered into orthogroups using 
+    // the best inflation parameter. 
+    ch_blastp = DIAMOND_BLASTP (
+        ch_orthofinder_full,
+        ch_dmd,
+        "txt",
+        "false",
+        []
     )
+    .txt
 
     //
-    // MODULE: MultiQC
+    // MODULE: Run Orthofinder's implementation of MCL (with similarity score
```

Comment:
I agree - the reason for doing it this way is because of how Orthofinder normalizes the e-values prior to MCL clustering. That's going to be a significant lift, so for now we're doing it this way (so we can make progress inferring the actual trees), but we can definitely discuss re-implementing later.

Reply:
what you could do is make an issue for it in this repo and add the tag "enhancement" so there's documentation of it

Reply:
Okay, I can do that!